### PR TITLE
dev-java/batik: Bump to 1.14 | CVE-2018-8013 CVE-2019-17566 CVE-2020-11987 

### PR DESCRIPTION
**Pull request by vaukai**
@Flowdalic

### CODE REVIEW

**Comment by vaukai** [comment]
The package has now 33 jar files with 7 of them having a MainClass and 4 of them not having any java classes.

To be done:
- Launchers
- Tests

**Comment by vaukai** [comment]
Tests still tbd.

**Comment by vaukai** [comment]
Most tricky part is the `*.policy` files. But at least `batik-squiggle-1.14` can now start without hickups.
- Tests still tbd.
- `batik-rasterizer-1.14` throws `access denied ("java.util.PropertyPermission" "java.security.policy" "write")`

**Comment by vaukai** [comment]
Most tricky part is the `*.policy` files. But at least `batik-squiggle-1.14` can now start without hickups.
But `batik-rasterizer-1.14` throws:
```
Exception in thread "main" java.security.AccessControlException: access denied ("java.util.PropertyPermission" "java.security.policy" "write")
        at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
        at java.base/java.security.AccessController.checkPermission(AccessController.java:897)
        at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
        at java.base/java.lang.System.setProperty(System.java:894)
        at org.apache.batik.util.ApplicationSecurityEnforcer.installSecurityManager(ApplicationSecurityEnforcer.java:257)
        at org.apache.batik.util.ApplicationSecurityEnforcer.enforceSecurity(ApplicationSecurityEnforcer.java:173)
        at org.apache.batik.apps.rasterizer.Main.execute(Main.java:938)
        at org.apache.batik.apps.rasterizer.Main.main(Main.java:1007)
```

**Comment by Leo3418** [comment]
> Most tricky part is the `*.policy` files. But at least `batik-squiggle-1.14` can now start without hickups. But `batik-rasterizer-1.14` throws:
> 
> ```
> Exception in thread "main" java.security.AccessControlException: access denied ("java.util.PropertyPermission" "java.security.policy" "write")
>         at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:472)
>         at java.base/java.security.AccessController.checkPermission(AccessController.java:897)
>         at java.base/java.lang.SecurityManager.checkPermission(SecurityManager.java:322)
>         at java.base/java.lang.System.setProperty(System.java:894)
>         at org.apache.batik.util.ApplicationSecurityEnforcer.installSecurityManager(ApplicationSecurityEnforcer.java:257)
>         at org.apache.batik.util.ApplicationSecurityEnforcer.enforceSecurity(ApplicationSecurityEnforcer.java:173)
>         at org.apache.batik.apps.rasterizer.Main.execute(Main.java:938)
>         at org.apache.batik.apps.rasterizer.Main.main(Main.java:1007)
> ```

I've figured it out!
```diff
diff --git a/dev-java/batik/batik-1.14.ebuild b/dev-java/batik/batik-1.14.ebuild
index 2670261c1..963f65062 100644
--- a/dev-java/batik/batik-1.14.ebuild
+++ b/dev-java/batik/batik-1.14.ebuild
@@ -186,6 +186,17 @@ src_install() {
 
 	done < <(grep -v '\(batik-rasterizer-ext\|batik-squiggle-ext\|batik-test-old\)' "${FILESDIR}"/batik-1.14-modules)
 
+	local java_policy_file="${JAVA_PKG_SHAREPATH}/etc/${PN}.policy"
+	insinto "$(dirname "${java_policy_file}")"
+	newins - "$(basename "${java_policy_file}")" <<- _EOF_
+	grant codeBase "file:${EPREFIX}${JAVA_PKG_JARDEST}/-" {
+	  permission java.security.AllPermission;
+	};
+	_EOF_
+	java-pkg_register-environment-variable \
+		gjl_java_args \
+		"-Djava.security.policy=file:${EPREFIX}${java_policy_file}"
+
 	if use doc; then
 		java-pkg_dojavadoc target/api
 	fi
```

File: dev-java/batik/files/batik-squiggle.desktop [hunk 1]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
+Name=Squiggle
+Comment=SVG browser
+Exec=batik-squiggle
```
Comment (Leo3418) [inline dev-java/batik/files/batik-squiggle.desktop:4]: Is this desktop entry supposed to show up in the applications menu?  I cannot find it after merging the ebuild

Comment (vaukai) [reply dev-java/batik/files/batik-squiggle.desktop]: @Leo3418 
Thanks for your help. `/usr/share/${PN}/etc/batik.policy` finally fixes it.
And I also added the slot to the launcher.

**Pull request closed without merging**